encoding/plain/plain.go

```diff
@@ -120,6 +120,7 @@ func (e *Encoding) DecodeDouble(dst []float64, src []byte) ([]float64, error) {
 func (e *Encoding) DecodeByteArray(dst []byte, src []byte, offsets []uint32) ([]byte, []uint32, error) {
 	dst, offsets = dst[:0], offsets[:0]
```

Comment:
This chunk shows that we are reusing `dst` and `offsets` passed to the method. Extending capacity of these slices creates new slices which negates the whole point of the PR.

I think, providing a method/function that calculates the anticipated capacity for the slices and leave  preallocating the `dst` and `offsets` slices to the caller of `(*Encoding)DecodeByteArray` will be more pragmatic.

I am suggesting something similar to  what `base64` package does except we fallback to vanilla `append` when the slices have smaller capacity.

`base64` has `(*Encoding) DecodedLen` and  `(*Encoding)Decode`. 

_`slices` package has been my swiss army knife to growing slices_

Reply:
Please ignore my comment. I just realised this is POC linked to a separate discussion that I was not aware of.

✌️